src/app/features/notification/model/notification-config.ts

```diff
@@ -0,0 +1,28 @@
+import { Entity } from "../../../core/entity/model/entity";
+import { DatabaseField } from "../../../core/entity/database-field.decorator";
+import { DatabaseEntity } from "../../../core/entity/database-entity.decorator";
+
+/**
+ * This represents one specific notification config for one specific user,
+ */
+@DatabaseEntity("NotificationConfig")
+export class NotificationConfig extends Entity {
+  @DatabaseField() channels: { [key: string]: boolean };
+
+  @DatabaseField() notificationTypes: NotificationType[];
+
+  constructor(id = null) {
+    super(id);
+  }
+}
+
+/**
+ * Represents a specific notification type configuration.
+ */
+export class NotificationType {
+  @DatabaseField() notificationType: string;
+  @DatabaseField() enabled: boolean;
+  @DatabaseField() channels: { [key: string]: boolean };
+  @DatabaseField() entityType: string;
+  @DatabaseField() conditions: any;
```

Comment:
```suggestion
  @DatabaseField() conditions: DataFilter;
```

(see type already defined in the code base and used for list filters)